orchestrator/gbt/src/client/deploy_erc20_representation.rs

```diff
@@ -99,9 +108,55 @@ pub async fn deploy_erc20_representation(
                     exit(0);
                 }
 
-                if Instant::now() - start > Duration::from_secs(100) {
-                    info!("Your ERC20 contract was not adopted, double check the metadata and try again");
-                    exit(1);
+                // we wait for up to WAIT_TIME seconds after that we must investigate why the attestation failed
+                const WAIT_TIME: u64 = 600;
+                if Instant::now() - start > Duration::from_secs(WAIT_TIME) {
+                    let attestations = grpc
+                        .get_attestations(QueryAttestationsRequest {
+                            limit: 0,
+                            order_by: String::new(),
+                            claim_type: String::new(),
+                            nonce: 0,
+                            height: 0,
+                        })
+                        .await;
+                    match attestations {
+                        Ok(attestations) => {
+                            let attestations = attestations.into_inner().attestations;
+                            for a in attestations {
+                                // the else condition here should never happen as it would mean we have an event with a nil pointer
+                                if let Some(claim) = a.claim {
+                                    if claim.type_url.contains("MsgERC20DeployedClaim") {
+                                        // decode any value to get at the actual contents of this claim
+                                        let mut buf = BytesMut::with_capacity(claim.value.len());
```

Comment:
I feel like we could use macros for interacting with Msgs and Anys, but that's something to consider later.

Reply:
We probably can I've just never sorted it out.